Engines/Wine/Plugins/virtual desktop/script.js

```diff
@@ -7,6 +7,8 @@ include("engines.wine.engine.object");
  * @returns {Wine} Wine object
  */
 Wine.prototype.setVirtualDesktop = function (width, height) {
+    const width = Bean("screenManager").getScreenWidth();
```

Comment:
Can you please move the direct `Bean(...)` access to a new utility script, like in https://github.com/PhoenicisOrg/scripts/blob/master/Utils/Functions/Filesystem/Files/script.js

This change makes it much easier later on to track the `Bean(...)` accesses and to ensure that no strange effect occur (see #1070).

Reply:
I don't get this.

Reply:
This will also override the sended with and height. So you should better do
```js
if (width === 'undefined')
    width = Bean("screenManager").getScreenWidth();
```
So that you take the current screen width if user does not provide a custom width for the virtual desktop.

Reply:
@ImperatorS79 even better is:

```javascript
if (!width) {
   width = ....
}
```

because this will also trigger if `width === null`

Reply:
@Zemogiter I basically suggest to implement a similar script to https://github.com/PhoenicisOrg/scripts/blob/master/Utils/Functions/Filesystem/Files/script.js with the difference that it proxies the `screenManager` Bean methods.

Reply:
I tried but I ended up with 2 scripts doing basically the same thing.

Reply:
What do you mean by doing the same? By splitting the Bean access to a new script you should have two scripts:
- one script functioning as a wrapper for the `screenManager` Bean. It should contain functions looking like:
   ```javascript
   function getScreenWidth() {
      return screenManager.getScreenWidth();
   }
   ```
- a second script containing the main logic. This script should look similar to:
   ```javascript
   Wine.prototype.setVirtualDesktop = function (width, height) {
      if (!width) {
         width = getScreenWidth();
      }
      if (!height) {
         height = getScreenHeight();
      }

      const regeditFileContent =
         "REGEDIT4\n" +
         "\n" +
         "[HKEY_CURRENT_USER\\Software\\Wine\\Explorer\\Desktops]\n" +
         "\"Default\"=\"" + width + "x" + height + "\"\n" +
         "[HKEY_CURRENT_USER\\Software\\Wine\\Explorer]\n" +
         "\"Desktop\"=\"" + "Default" + "\"\n";

      this.regedit().patch(regeditFileContent);

      return this;
   }; 
   ```

Reply:
Done.